tests/test_resolver.py

```diff
@@ -26,3 +50,99 @@ def test_open_stage_and_logging(capfd):
+    )
+
+    assert_parking_lot_structure(stage)
+
+
+# Given a USD document that references an asset via a relative search-path
+# based file path, then the asset is resolved the the file path as expected.
+def test_openassetio_resolver_has_no_effect_with_search_path():
+
+    context = build_search_path_context(
+        "resources/integration_test_data/resolver_has_no_effect_with_search_path/search_path_root"
+    )
+    stage = Usd.Stage.Open(
+        "resources/integration_test_data/resolver_has_no_effect_with_no_search_path/parking_lot.usd",
+        context,
+    )
+
+    assert_parking_lot_structure(stage)
+
+
+# Given a USD document that recursively references multiple other USD
+# documents, and expects the resolve output structured in the form of
+# adjacent relative file paths, then each document can be resolved via
+# OpenAssetIO.
+@pytest.mark.xfail(reason="OpenAssetIO not integrated yet")
+def test_recursive_resolve_to_adjacent_file_path():
+    stage = Usd.Stage.Open(
+        "resources/integration_test_data/recursive_resolve_to_adjacent_file_path/parking_lot.usd"
+    )
+
+    assert_parking_lot_structure(stage)
+
+
+# Given a USD document that recursively references multiple other USD
+# documents, and expects the resolve output structured in the form of
+# search-path based file paths, then each document can be resolved via
+# OpenAssetIO.
+@pytest.mark.xfail(reason="OpenAssetIO not integrated yet")
+def test_recursive_resolve_to_search_path():
+    context = build_search_path_context(
+        "resources/integration_test_data/recursive_resolve_to_search_path/search_path_root"
+    )
+
+    stage = Usd.Stage.Open(
+        "resources/integration_test_data/recursive_resolve_to_search_path/parking_lot.usd",
+        context,
+    )
+
+    assert_parking_lot_structure(stage)
+
+
+# Given a USD document that references a second level document via an
+# assetized reference resolvable by OpenAssetIO, and that second level
+# document containing a non-assetized, adjacent relative file path
+# reference to a third level document, then the document can be fully
+# resolved.
+@pytest.mark.xfail(reason="OpenAssetIO not integrated yet")
+def test_assetized_child_ref_non_assetized_grandchild():
+    stage = Usd.Stage.Open(
+        "resources/integration_test_data/assetized_child_ref_non_assetized_grandchild/parking_lot.usd"
+    )
+
+    assert_parking_lot_structure(stage)
+
+
+# Given a USD document that references a second level document via an
+# non-assetized, adjacent relative file path reference, and that second
+# level document containing an assetized reference resolvable by
+# OpenAssetIO to a third level document, then the document can be fully
+# resolved.
+@pytest.mark.xfail(reason="OpenAssetIO not integrated yet")
+def test_non_assetized_child_ref_assetized_grandchild():
+    stage = Usd.Stage.Open(
+        "resources/integration_test_data/non_assetized_child_ref_assetized_grandchild/parking_lot.usd"
```

Comment:
[fix] We're gonna have to do some `pylint`ing and `black`ing, to avoid having to fixup once the CI linting PR is merged.

Reply:
https://github.com/OpenAssetIO/usdOpenAssetIOResolver/pull/13/commits/70e43c109ea708191783e554ad0b07b0af461653

pylint and black changes. Did it manually so possible the merged lint changes could still catch some things.